server/policy.go

```diff
@@ -13,7 +19,69 @@ import (
 // will describe in what way the state failed. See the Policy documentation for
 // more information about the specifics of different policies.
 func EvaluatePolicy(state *pb.MachineState, policy *pb.Policy) error {
-	return evaluatePlatformPolicy(state.GetPlatform(), policy.GetPlatform())
+	if err := evaluatePlatformPolicy(state.GetPlatform(), policy.GetPlatform()); err != nil {
+		return err
+	}
+	if state.GetTeeAttestation() == nil {
+		return nil
+	}
+	switch at := state.TeeAttestation.(type) {
+	case *pb.MachineState_SevSnpAttestation:
+		return evaluateSevSnpPolicy(at.SevSnpAttestation, policy.GetSevSnp())
+	case *pb.MachineState_TdxAttestation:
+		// Currently unchecked.
+		return nil
+	default:
+		return fmt.Errorf("no policy for TEE attestation type %T", state.TeeAttestation)
+	}
+}
+
+func rootOfTrust(certs [][]byte) (*x509.CertPool, error) {
+	pool := x509.NewCertPool()
+	for _, der := range certs {
+		cert, err := x509.ParseCertificate(der)
+		if err != nil {
+			return nil, err
+		}
+		pool.AddCert(cert)
+	}
+	return pool, nil
+}
+
+// the SEV-SNP attestation signature is already verified by this point.
+func evaluateSevSnpPolicy(state *spb.Attestation, policy *pb.SevSnpPolicy) error {
+	if policy == nil {
```

Comment:
This feels like a scary way to represent any "everything" policy to me. I feel like this is really easily done accidentally.
I don't know enough about protocol buffers to have a suggestion here, so perhaps when we have a different sort of policy language this concern will no longer be relevant.
WDYT?

Reply:
policy == nil should be understood as &pb.SevSnpPolicy{}. I've removed this check. The real concern is the `policy.GetUefi() == nil` check, since if I treat nil as &policy.RIMPolicy{}, then the behavior is different.

I can change the behavior to be like the evaluatePlatformPolicy, where at least one root has to be present to care about any signers. No signers means any value is okay. That would make the nil check and empty message the same.

I wouldn't want to start failing all policies because there are no signed measurements deployed yet. Do you want to depend on signed reference values right now? I suppose you could but then have failing tests internally until the signed firmware package finally rolls out.